File: src/Application Specific/Classes/Entity Collections/VATRates/VATRatesValidator.vb
Namespace Entity

Namespace VATRatess

Public Class VATRatesValidator
    'make sure that contact object is valid
    Inherits BaseValidator

    Public Sub Validate(ByVal oVATRates As VATRates)

        'make sure that contact object is valid
        If oVATRates.Errors.Count > 0 Then
            Dim de As DictionaryEntry
            For Each de In oVATRates.Errors
                Me.AddCriticalMessage(CStr(de.Value))
            Next
        End If

                If oVATRates.VATRateCode.Trim.Length = 0 Then
                    Me.AddCriticalMessage("Code Missing")
                End If
	
        If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
            Throw New ValidationException(Me)
        End If

    End Sub

End Class

End Namespace

End Namespace

File: apollo/apollo.fsm/src/Application Specific/Classes/Entity Collections/VATRates/VATRatesValidator.cs
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace VATRatess
    {
        // make sure that contact object is valid
        public class VATRatesValidator : BaseValidator
        {
            public void Validate(VATRates oVATRates)
            {
                // make sure that contact object is valid
                if (oVATRates.Errors.Count > 0)
                {
                    foreach (DictionaryEntry de in oVATRates.Errors)
                        AddCriticalMessage(Conversions.ToString(de.Value));
                }

                if (oVATRates.VATRateCode.Trim().Length == 0)
                {
                    AddCriticalMessage("Code Missing");
                }

                if (ValidatorMessages.CriticalMessages.Count > 0)
                {
                    throw new ValidationException(this);
                }
            }
        }
    }
}